### PR TITLE
Monthly radio button. FirstTry

### PR DESCRIPTION
**Pull request by Pitcity**
added functionality on monthly button with possibility to edit events

### CODE REVIEW

File: src/main/webapp/resources/js/renderCalendar.js [hunk 1]
```diff
@@ -188,9 +196,16 @@ $(function () {
     });
 
     $('#update-recurrent-button').click(function () {
-        if(validateEventDialogData(UPDATE_RECURRENT_EVENT)){
```
Comment (Pitcity) [inline src/main/webapp/resources/js/renderCalendar.js]: checking that validation for each type of radio-button. inside validateEventDialogData for CREATE_RECURRENT_EVENT && UPDATE_RECURRENT_EVENT we have the same code. and actually updating is deleting existent event + creating new one

File: src/main/java/ua/softserveinc/tc/service/impl/CalendarServiceImpl.java [hunk 2]
```diff
@@ -131,16 +132,77 @@ public void add(Event event) {
...
+        Set <Integer> daysOfTheMonth = new HashSet<>();
         Calendar calendar = Calendar.getInstance();
         for (Event event : listOfRecurrentEvent) {
             calendar.setTime(event.getStartTime());
             weekDays.add(calendar.get(Calendar.DAY_OF_WEEK));
+            daysOfTheMonth.add(calendar.get(Calendar.DAY_OF_MONTH));
+        }
+        if (listOfRecurrentEvent.get(0).getRecurrentType().equals("Monthly")) {
+            int[] tempDays = new int[daysOfTheMonth.size()];
+            int index =0;
```
Comment (VitalijF) [inline src/main/java/ua/softserveinc/tc/service/impl/CalendarServiceImpl.java]: int index = 0;

File: src/main/webapp/resources/js/renderCalendar.js [hunk 3]
```diff
@@ -695,7 +762,25 @@ function clearEventDialogSingleMulti(){
...
+           for(var j = 0; j<DAYS_IN_WEEK;j++) {
+               dayNum++;
+               if (dayNum>DAYS_IN_MONTH) break;
+               tableHTML+=("<td>" + (dayNum) + "</td>");
+           }
+        tableHTML+="</tr>";
+    }
+    monthlyTable.innerHTML=tableHTML;
+    $('table').find('td').click(function() {
+        $(this).toggleClass("active");
```
Comment (VitalijF) [inline src/main/webapp/resources/js/renderCalendar.js]: why you mix js and jquery
document.getElementById("monthly-days") -> use $('#monthly-days)

File: src/main/webapp/resources/js/renderCalendar.js [hunk 4]
```diff
@@ -695,7 +762,25 @@ function clearEventDialogSingleMulti(){
     $('#color-select').val("#1ba1e2");
     $('#event-title').val("");
     $('#description').val("");
+}
 
+function buildTableMonthly() { //generating table for DAYS_IN_MONTH day
+    var tableHTML = '';
+    var dayNum = 0;
+    var monthlyTable = document.getElementById("monthly-days");
```
Comment (VitalijF) [inline src/main/webapp/resources/js/renderCalendar.js]: Why you mix js and jquery
use $("#monthly-days") instead document.getElementById("monthly-days");

Comment (Pitcity) [inline src/main/webapp/resources/js/renderCalendar.js]: doesnt work that way

**Pull request merged**